src/hotspot/share/services/diagnosticArgument.cpp

```diff
@@ -350,3 +354,31 @@ template <> void DCmdArgument<MemorySizeArgument>::init_value(TRAPS) {
 }
 
 template <> void DCmdArgument<MemorySizeArgument>::destroy_value() { }
+
+template <>
+void DCmdArgument<FileArgument>::parse_value(const char *str, size_t len,
+                                                   TRAPS) {
+  if (str == NULL) {
+    _value._name = nullptr;
+  } else {
+    _value._name = NEW_C_HEAP_ARRAY(char, JVM_MAXPATHLEN, mtInternal);
+    if (!Arguments::copy_expand_pid(str, len, _value._name, JVM_MAXPATHLEN)) {
+      fatal("Invalid file path: %s", str);
```

Comment:
As I understand the 'copy_expand_pid' might fail if very long line is used. This cause jvm crash.,
So there is possibility that user might crash jvm accidentally invoking jcmd command. 
It doesn't look safe, I believe it would be better to throw Exception like for any other invalid command, see
" THROW_MSG(vmSymbols::java_lang_IllegalArgumentException(),"

The 'fatal" owuld make sense only if failing of 'copy_expand_pid' means some unrecoverable jvm bug.

Reply:
If the attach listener thread doesn't actually need to be a Java thread then you could look into changing that. Not sure it would really buy us that much in terms of added robustness though.